frontend/packages/core/src/time.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import type { ReactTimeagoProps as TimeAgoProps } from "react-timeago";
+import TimeAgo from "react-timeago";
+
+interface EventTimeProps extends Pick<TimeAgoProps, "date"> {
+  onClick?: () => void;
+  short?: boolean;
+}
+
+const unitFormatter = (unit: string): string => {
+  switch (unit) {
+    case "month":
+      // month -> mo
+      return unit.substring(0, 2);
+    default:
+      return unit.charAt(0);
+  }
+};
+
+const parseTimestamp = (timestamp?: number | Long | null): number => {
+  return parseInt(timestamp?.toString() || "0", 10);
+};
+
+const setMilliseconds = (timestamp?: number | Long | null): number => {
+  const ts = new Date(0);
+  return ts.setUTCMilliseconds(parseTimestamp(timestamp));
+};
+
+const EventTime = ({ short = true, onClick, ...props }: EventTimeProps) => (
```

Comment:
Should we call this something like `TimeAgo` or `ElapsedTime`? `EventTime` seems a bit generic and doesn't denote that it has a live count.

Reply:
Yeah TimeAgo makes sense, will update